config/templates/elasticsearch-db.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: templating.flanksource.com/v1
+kind: Template
+metadata:
+  name: elasticsearch-db
+spec:
+  source:
+    apiVersion: db.flanksource.com/v1
+    kind: ElasticsearchDB
+  resources:
+    - apiVersion: elasticsearch.k8s.elastic.co/v1
+      kind: Elasticsearch
+      metadata:
+        name: logs
+        namespace: '{{ .metadata.namespace }}'
+      spec:
+        version: '{{ .spec.version }}'
+        nodeSets:
+          - name: default
+            count: '{{ .spec.replicas | default 3 }}'
+            config:
+              node.master: true
+              node.data: true
+              node.ingest: true
+              node.store.allow_mmap: false
+              xpack.security.transport.ssl.supported_protocols: TLSv1.1,TLSv1.2
+              xpack.security.authc.anonymous.roles: fluentd
+            podTemplate:
+              spec:
+                containers:
+                  - name: elasticsearch
+                    env:
+                      - name: ES_JAVA_OPTS
+                        value: '-Xms{{ .spec.heap | default "2Gi" | strings.ReplaceAll "Ki" "k" | strings.ReplaceAll "Mi" "m" | strings.ReplaceAll "Gi" "g"  }} -Xmx{{ .spec.mem.requests | default "2Gi" | strings.ReplaceAll "Ki" "k" | strings.ReplaceAll "Mi" "m" | strings.ReplaceAll "Gi" "g" }}'
+                    resources:
+                      requests:
+                        memory: '{{ .spec.resources.requests.memory | default "2Gi" }}'
+                        cpu: 0.5
```

Comment:
Can we template and default these aswell ?

Reply:
The cpu values?  Or do you want

resources: '{{ .spec.resources | default coll.Dict() }}'

Reply:
e.g. `cpu: '{{ .spec.resources.requests.cpu | default "500m" }}' `

i.e. all requests/limits should be configurable with defaults